### ISSUE TITLE
[Bug] SingleValue component does not run onClick in mobile view

### ISSUE THREAD

**Issue by s7dhansh** [JedWatson/react-select#4000]
https://codesandbox.io/embed/react-codesandboxer-example-bfv06?codemirror=1

I have put a simple onClick on a custom SingleValue component. The onClick is fired in normal browser view, as seen by console logs.

Steps to reproduce:
1. Open devtools and switch to mobile view
2. Try clicking on the single value component now. No logs in console.

I am guessing the event is obscured by some other top level event.

**Comment by flibertigibet** [comment]
Experiencing the same issue with **components.DropdownIndicator**.

Wrapped it in a div with an onClick and it just doesn't fire in mobile view. There are no error logs either.

**Comment by johnfrommartin** [comment]
> I am guessing the event is obscured by some other top level event.

@s7dhansh You're correct. The root issue that both of you are experiencing is that `components.SingleValue` has default event handlers that intercept the events of the children node(s). 

Took me longer than I expected but you'll need to either:
1. override the default behavior around the `<components.SingleValue>` component. 
2. build your own `<components.SingleValue>` component from the bottom up

Here's what my custom `SingleValue` component looks like, using the 2nd approach:
```
const {
  getStyles,
  selectProps: { ref, ...restSelectProps },
} = props;
SingleValueRender = (
  <div
    {...restSelectProps}
    ref={ref}
    style={getStyles('singleValue', props)}
    onMouseDown={e => { // ! override default onMouseDown
      e.stopPropagation();
      e.preventDefault();
    }}
  >
    {props.children}
    <button
      onClick={props.clearValue}
    >
      <IconClearValue />
    </button>
  </div>
);
```

I found this solution straight from a similar closed GH issue #3117, where the solve is demonstrated in this sandbox @ https://codesandbox.io/s/react-codesandboxer-example-fmxdi

**Comment by s7dhansh** [comment]
@johnfrommartin your sandbox does not work. Kindly check it in simulated mobile view.

**Comment by bryankenote** [comment]
@s7dhansh have you tried passing your click handler to both `onMouseUp` and `onTouchEnd`.

**Comment by s7dhansh** [comment]
@bryankenote thank you!
`onTouchEnd` works. Updated the sandbox to showcase the same.
